src/sema/mutability.rs

```diff
@@ -27,9 +27,15 @@ pub fn mutability(file_no: usize, ns: &mut Namespace) {
 /// While we recurse through the AST, maintain some state
 struct StateCheck<'a> {
     diagnostics: Vec<Diagnostic>,
+    /// Does the function have an expression or statement that reads from state?
     does_read_state: bool,
+    /// Does the function have an expression or statement that writes to state?
     does_write_state: bool,
+    /// Does the function have an expression or statement that may write to state?
+    may_write_state: bool,
+    /// Is the function declared view, payer or default mutability
     can_read_state: bool,
+    /// Is the function declared with default or payer mutability
```

Comment:
You used the question mark on the other questions, but not here. Please, adopt a single notation.

Reply:
I've added question marks but I think it questionable whether this improves things.